src/Dropdown/index.js

```diff
@@ -0,0 +1,195 @@
+import React, { Children, PureComponent } from 'react';
+import PropTypes from 'prop-types';
+import posed, { PoseGroup } from 'react-pose';
+import styled from 'styled-components';
+import onClickOutside from 'react-onclickoutside';
+
+import SearchBar from './SearchBar';
+import { Header, Menu, MenuItem, SearchInputContainer } from './elements';
+
+const { bool, func, node, string } = PropTypes;
+
+/**
+ * Dropdown
+ *
+ * This component is in charge of displaying
+ * a dropdown
+ *
+ * @param {node} children // Anything that can be rendered: numbers, strings, elements or an array (or fragment).
+ * @param {string} className // Add a text aside in the select next the selected value.
+ * @param {node} noresultLabel // Label to display when no result found.
+ * @param {bool} searchable // Whether the dropdown is searchable.
+ * @param {string} searchBarPlacholder // SearchBar input placholder.
+ * @param {node} title // Dropdown title.
+ *
+ * @return {jsx}
+ */
+
+class Dropdown extends PureComponent {
+  /** Prop types. */
+  static propTypes = {
+    children: node.isRequired,
+    className: string,
+    noResultLabel: string,
+    onToggle: func,
+    searchable: bool,
+    searchBarPlacholder: string,
+    title: node.isRequired,
+  };
+
+  /** Default props. */
+  static defaultProps = {
+    className: '',
+    noResultLabel: null,
+    onToggle: () => {},
+    searchable: false,
+    searchBarPlacholder: '',
+  };
+
+  /** Internal state. */
+  state = {
+    displayMenu: false,
+    searchKeyword: '',
+  };
+
+  /** preserve the initial state in a new object. */
+  baseState = this.state;
+
+  /**
+   * Toogle Menu
+   */
+  toggleMenu = () => {
+    const { displayMenu } = this.state;
+    const { onToggle } = this.props;
+
+    this.setState(
+      prevState => ({
+        displayMenu: !prevState.displayMenu,
+      }),
+      () => {
+        // Clear input value if menu is open
+        if (displayMenu) {
+          this.resetSearchInput();
+        }
+        onToggle && onToggle(!displayMenu);
+      },
+    );
+  };
+
+  /**
+   * Handle Click Outside
+   */
+  handleClickOutside = () => {
```

Comment:
Why is there a different logic than in `toggleMenu`?
Why don't you refactor both methods?

Reply:
@leopoldhoudin I use [react-onclickoutside](https://github.com/Pomax/react-onclickoutside) to handle the outside click correctly. We need to put  `handleClickOutside` method to work like explain in the module documentation.

Reply:
Okay, yet you don't call `onToggle` prop here, which should be called. 

When I was talking about refactoring, I was thinking about something as follows:
```jsx
toggleMenu = displayMenu => {
  const { onToggle } = this.props;

  this.setState({ ...this.state, displayMenu }, () => {
    if (displayMenu) this.resetSearchInput();

    onToggle && onToggle(displayMenu);
  });
};

handleHeaderClick = () => {
  const { displayMenu } = this.state;
  this.toggleMenu(!displayMenu);
};

handleClickOutside = () => {
  this.toggleMenu(false);
};

/* ... */

render() {
  /* ... */

   <Header
        onClick={this.handleHeaderClick}
        aria-haspopup="true"
        aria-expanded={displayMenu}
        data-test="dropdown-header"
      >

/* ... */
  
```

Reply:
@leopoldhoudin Oh I see, good point !